packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -264,7 +265,10 @@ export function createVM(elm: HTMLElement, Ctor: ComponentConstructor, options:
 
     // link component to the wire service
     const initializedVm = uninitializedVm as VM;
-    linkComponent(initializedVm);
+    // initializing the wire decorator per instance only when really needed
```

Comment:
move this to the lightning element constructor.

Reply:
When we install the wire adapters, we need to call the config function that will pull the data from the component. If we move this to the element constructor, those values are not yet initialized, therefore we will get the assertion message.